File: app/src/main/java/net/daverix/ajvm/jvm/AALoadOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class AALoadOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int index = (int) currentFrame.pop();
        Object[] array = (Object[]) currentFrame.pop();
        currentFrame.push(array[index]);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/AALoadOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.ByteCodeReader
import net.daverix.ajvm.Frame
import java.io.IOException

class AALoadOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val index = currentFrame.pop() as Int
        val array = currentFrame.pop() as Array<Any?>
        currentFrame.push(array[index])
    }
}
